File: Town_judge.cpp
// Problem Link https://leetcode.com/problems/find-the-town-judge/description/

class Solution
{
public:
    int findJudge(int n, vector<vector<int>> &trust)
    {
        if (n == 1)
        {
            return 1;
        }

        unordered_map<int, int> mp;
        vector<int> possibles;
        for (int i = 0; i < trust.size(); i++)
        {
            mp[trust[i][1]]++;
        }

        for (auto itr : mp)
        {
            if (itr.second == n - 1)
            {
                possibles.push_back(itr.first);
            }
        }

        for (int i = 0; i < possibles.size(); i++)
        {
            int found = 0;
            for (int j = 0; j < trust.size(); j++)
            {
                if (trust[j][0] == possibles[i])
                {
                    found = 1;
                    break;
                }
            }
            if (found == 0)
            {
                return possibles[i];
            }
        }
        return -1;
    }
};

File: town_judge.py
class Solution:
    def findJudge(self, N: int, trust: List[List[int]]) -> int:
        if N == 1:
            return 1

        trusts = [0] * (N + 1)
        istrusted_by = [0] * (N + 1)

        for a, b in trust:
            trusts[a] += 1
            istrusted_by[b] += 1

        for i in range(1, N + 1):
            if trusts[i] == 0 and istrusted_by[i] == N - 1:
                return i
        return -1